scipy-stubs/signal/_savitzky_golay.pyi

```diff
@@ -1,20 +1,27 @@
-from scipy._typing import Untyped, UntypedArray
+from typing import Literal, TypeAlias
+
+import numpy as np
+import numpy.typing as npt
+import optype as op
+
+_Array_fc_1d: TypeAlias = np.ndarray[tuple[int], np.dtype[np.inexact[npt.NBitBase]]]
+_Mode: TypeAlias = Literal["mirror", "constant", "nearest", "wrap", "interp"]
 
 def savgol_coeffs(
     window_length: int,
     polyorder: int,
     deriv: int = 0,
     delta: float = 1.0,
     pos: int | None = None,
-    use: str = "conv",
-) -> UntypedArray: ...
+    use: Literal["conv", "dot"] = "conv",
+) -> _Array_fc_1d: ...
```

Comment:
I can't find anything in the docs about complex return types. So I'm missing something, then using `np.floating[Any]` instead here would be best.

Reply:
Thanks for the heads up. I think I just saw that the output was due to some sort of convolution and took the return type from there. I don't think the coefficients of the filter can be complex but I'm also not too well versed on filters.

Reply:
Should be fixed now.